spec/models/submission_spec.rb

```diff
@@ -2,6 +2,40 @@
 
 RSpec.describe Submission do
 
+  it { should validate_presence_of(:user) }
+  it { should validate_presence_of(:submission_type) }
+
+  describe '#submission_type' do
+    let(:submission) { create(:submission) }
+
+    it 'allows only certain submission type values' do
+      %w(population traits qtl linkage_map).each do |t|
+        submission.submission_type = t
+        expect(submission.save).to be_truthy
```

Comment:
I do not think it cheats because I believe that AR object should behave nicely even if they are not saved yet ;)